v2/dtos/requests/subscription_test.go

```diff
@@ -0,0 +1,298 @@
+//
+// Copyright (C) 2020-2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testSubscriptionName       = "subscriptionName"
+	testSubscriptionCategories = []string{models.SoftwareHealth}
+	testSubscriptionLabels     = []string{"label"}
+	testSubscriptionChannels   = []dtos.Channel{
+		{Type: models.Email, EmailAddresses: []string{"test@example.com"}},
+	}
+	testSubscriptionDescription    = "description"
+	testSubscriptionReceiver       = "receiver"
+	testSubscriptionResendLimit    = int64(5)
+	testSubscriptionResendInterval = "10s"
+	unsupportedChannelType         = "unsupportedChannelType"
+	unsupportedCategory            = "unsupportedCategory"
+)
+
+func addSubscriptionRequestData() AddSubscriptionRequest {
+	return AddSubscriptionRequest{
+		BaseRequest: common.BaseRequest{
+			RequestId: ExampleUUID,
```

Comment:
Add setting of ApiVersion now that is is required. Example here:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/requests/device_test.go#L40

Reply:
Added.